### PR TITLE
Initial Co-occurence plots  

### PR DESCRIPTION
**Pull request by jashapiro**
#### Purpose/implementation

Which pairs of genes are repeatedly mutated across tumors?

This script compiles lists of genes mutated in each sample, then looks for pairs of genes that are mutated together across samples more (or less) often than would be expected by chance. 
Analysis is on the gene level, so a single mutation in a gene counts the same as many mutations in the gene.
Also, current analysis excludes synonymous mutations and intron mutations, but includes other non-coding mutations.

Currently looks at all tumor types, and choses which genes to analyze based on finding the most mutated genes.

This is the basic outline of the analysis, but updates are expected to add tumor type based analysis, as well as specific analysis of driver genes.

#### Issue

#13 

#### Directions for reviewers

* Overall code organization? I could make things more modular, but wanted to get a first pass analysis out.
* Do the statistics seem correct and reasonable? 
* Should the co-occurence calculations be updated to include q-values in addition to raw p-values?

* For future work: 
  * Which gene sets should be evaluated?
  * Which types of mutations? Should this be run with only coding mutations?
  * Which Tumor groups?

#### Results

Table of co-occurence results and resulting figure that plots relationships between genes based on log10(p-values), with positive scores indicating co-occurence of mutations and negative being mutual exclusivity (which is not apparent in this initial analysis).

#### Docker and continuous integration

- [x] This analysis has been added to continuous integration.

### CODE REVIEW

File: analyses/interaction-plots/scripts/01-process_mutations.R [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    opt_str = "--out", type = "character", 
+    default = file.path("analyses", "interaction-plots", "results", "cooccurence.tsv"),
+    help = "Relative file path (from top directory of 'OpenPBTA-analysis')
+            where output table will be placed.",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--specimen_list", type = "character", 
+    default = file.path("analyses", "independent-samples", "results", "independent-specimens.wgs.primary.tsv"),
+    help = "Relative file path (from top directory of 'OpenPBTA-analysis')
```
Comment (jaclyn-taroni) [inline analyses/interaction-plots/scripts/01-process_mutations.R]: Looks like a copy paste from the MAF option.

File: analyses/interaction-plots/scripts/01-process_mutations.R [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  ),
+  make_option(
+    opt_str = "--out", type = "character", 
+    default = file.path("analyses", "interaction-plots", "results", "cooccurence.tsv"),
+    help = "Relative file path (from top directory of 'OpenPBTA-analysis')
+            where output table will be placed.",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--specimen_list", type = "character",
```
Comment (jaclyn-taroni) [inline analyses/interaction-plots/scripts/01-process_mutations.R]: Are you planning on leveraging this option to filter to disease types or would you split up the MAF data.frames based on something in the metadata? This is not important yet, but trying to understand where this is headed.

File: analyses/interaction-plots/scripts/02-plot_interactions.R [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+# Command line example:
+#
+# Rscript analyses/interaction-plots/02-plot_interactions.R \
+#   --infile 
+
+#### Initial Set Up
+# Establish base dir
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+script_root <- file.path(root_dir, "analyses", "interaction-plots")
+
```
Comment (jaclyn-taroni) [inline analyses/interaction-plots/scripts/02-plot_interactions.R]: You have a `root_dir` below and `script_root` is not used anywhere else.

File: analyses/interaction-plots/scripts/02-plot_interactions.R [hunk 4]
```diff
@@ -0,0 +1,82 @@
+# Create plot of co-occurence and mutual exclusivity
+#
+# JA Shapiro for ALSF - CCDL
+#
+# 2019
+#
+# Option descriptions
```
Comment (jaclyn-taroni) [inline analyses/interaction-plots/scripts/02-plot_interactions.R]: Documentation's incomplete here and in the command line example.

File: analyses/interaction-plots/scripts/cooccur_functions.R [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  # filter on vaf and consequence (for single Consequence cases)
+  maf_df <- maf_df %>% 
+    dplyr::filter(vaf >= min_vaf) %>%
+    dplyr::mutate(consequence_list = stringr::str_split(Consequence, ",")) %>% 
+    dplyr::rowwise() %>%
+    dplyr::filter(!all(unlist(consequence_list) %in% exclude_consequence))
+    
+  if (length(include_consequence) > 0){
+    maf_df <- maf_df %>% 
+      dplyr::filter(any(unlist(consequence_list) %in% exclude_consequence))
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply analyses/interaction-plots/scripts/cooccur_functions.R]: You think correctly. That's what I get for not testing...

File: analyses/interaction-plots/scripts/01-process_mutations.R (hunk 2 above)
Comment (jashapiro) [reply analyses/interaction-plots/scripts/01-process_mutations.R]: I hadn't yet decided on this. I think maybe generating the specimen lists in a separate script, then using them here with little change may be the way to go. Different types may also have different gene lists of interest, but I haven't plugged that in as a script level option yet. The main calculation function can handle arbitrary gene lists though, so it should be easy to add that.

File: analyses/interaction-plots/scripts/02-plot_interactions.R (hunk 4 above)
Comment (jashapiro) [reply analyses/interaction-plots/scripts/02-plot_interactions.R]: Will fix here, but does the command line example need all options?

Comment (jaclyn-taroni) [reply analyses/interaction-plots/scripts/02-plot_interactions.R]: Normally I would say no, but in this case it's an input file and an output file.

**Pull request merged**